clang/test/SemaSYCL/intel-max-work-group-size.cpp

```diff
@@ -59,3 +65,31 @@ void instantiate() {
   // expected-note@#f7prev {{previous attribute is here}}
   f7<2, 2, 2>(); // expected-note {{in instantiation}}
 }
+
+
+// If the [[intel::max_work_group_size(X, Y, Z)]] attribute is specified on
+// a declaration along with [[sycl::reqd_work_group_size(X1, Y1, Z1)]]
+// attribute, check to see if values of reqd_work_group_size arguments are
+// equal or less than values coming from max_work_group_size attribute.
+[[sycl::reqd_work_group_size(64, 64, 64)]]
+[[intel::max_work_group_size(16, 16, 16)]]  //expected-error{{'max_work_group_size' attribute conflicts with 'reqd_work_group_size' attribute}}
+void f9() {}
+
+[[intel::max_work_group_size(4, 4, 4)]] void f10();
+[[sycl::reqd_work_group_size(2, 2, 2)]] void f10(); // OK
+
+[[sycl::reqd_work_group_size(2, 2, 2)]]
+[[intel::max_work_group_size(4, 4, 4)]] void f11() {} // OK
+
+// FIXME: We do not have support yet for checking
+// reqd_work_group_size and max_work_group_size
+// attributes when merging, so the test compiles without
+// any diagnostic when it shouldn't.
+[[sycl::reqd_work_group_size(64, 64, 64)]] void f12();
```

Comment:
Can we fix this in this PR? Don't we just need to add similar logic in ReqdWorkGroupSizeAttr handlers?

Reply:
@elizabethandrews, does this answer your question?